vendor: bump pebble to pick up snappy bump

dt opened the pull request:
Bumps pebble to pick up snappy v0.0.3. Changes included:

- 0369dc2e bump snappy to v0.0.3
- 99939fea vfs: use Bavail in GetFreeSpace
- a9e4c56e vfs: add OnDiskFull VFS wrapper
- a7f14024 sstable: vectorize shared prefix byte comparison
- abf31e35 sstable: add shared prefix to BenchmarkWriter
- 91c52f60 compaction: hoist nilPacer check outside compaction loop

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/66549)
<!-- Reviewable:end -->

dt committed:
vendor: bump pebble to pick up snappy bump

Bumps pebble to pick up snappy v0.0.3. Changes included:
0369dc2e bump snappy to v0.0.3
99939fea vfs: use Bavail in GetFreeSpace
a9e4c56e vfs: add OnDiskFull VFS wrapper
a7f14024 sstable: vectorize shared prefix byte comparison
abf31e35 sstable: add shared prefix to BenchmarkWriter
91c52f60 compaction: hoist nilPacer check outside compaction loop

Release note: none.

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66549#-:-McL-UTW4MKxDpNNP88F:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @itsbilal)



<!-- Sent from Reviewable.io -->

dt commented:
TFTR!

bors r+
